### PR TITLE
Reserved keyword schema

### PR DESCRIPTION
**Pull request by bharadwaj-aditya**
Added handling to migrate tables with reserved keywords as table/column names in the database.

### CODE REVIEW

File: v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapperTest.java [hunk 1]
```diff
@@ -248,20 +248,6 @@ public void testJdbcIoWrapperDifferentTables() throws RetriableSchemaDiscoveryEx
     String testCol = "ID";
     SourceColumnType testColType = new SourceColumnType("INTEGER", new Long[] {}, null);
     when(mockDialectAdapter.discoverTables(any(), any())).thenReturn(ImmutableList.of("testTable"));
-    when(mockDialectAdapter.discoverTableIndexes(any(), any(), any()))
```
Comment (Deep1998) [inline v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapperTest.java:251]: Is this deletion expected? It looks important and we are not removing this for the other tests?

**Review by Deep1998 — DISMISSED** [review]

**Review by VardhanThigle — COMMENTED** [review]
Thanks for the detailed call.
As we had discussed, I am trying out setting ANSI SQL mode at session level to fix the issue addressed here at #1987 . \" was used in order to have a common handling across databases, which if it works, would make it simpler.

File: v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java [hunk 2]
```diff
@@ -361,7 +361,7 @@ protected static String delimitColumnName(String columnName) {
 
   private static PartitionColumn partitionColumnFromIndexInfo(SourceColumnIndexInfo idxInfo) {
     return PartitionColumn.builder()
-        .setColumnName(delimitColumnName(idxInfo.columnName()))
```
Comment (Deep1998) [inline v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java:364]: We can probably remove delimitColumnName if not used anymore

File: v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapperTest.java (hunk 1 above)
Comment (bharadwaj-aditya) [reply v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapperTest.java:251]: this was thrown up as an unused mock, so it has been removed.

**Comment by bharadwaj-aditya** [comment]
Closing this as the behaviour is captured via an alternate flow. Details in - https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1987

**Pull request closed without merging**